app/services/candidate_interface/sign_in_candidate.rb

```diff
@@ -0,0 +1,53 @@
+module CandidateInterface
+  class SignInCandidate
+    def self.call(email:, controller_context:)
+      new(email, controller_context).call
+    end
+
+    attr_reader :controller_context, :email
+
+    def initialize(email, controller_context)
+      @controller_context = controller_context
+      @email = email
+    end
+
+    delegate(
+      :params,
+      :redirect_to,
+      :render,
+      :candidate_interface_check_email_sign_in_path,
+      to: :controller_context,
+    )
+
+    def call
+      candidate = Candidate.for_email email
+
+      if candidate.persisted?
+        update_course_from_find(candidate)
+        MagicLinkSignIn.call(candidate: candidate)
+
+        controller_context.send :add_identity_to_log, candidate.id
```

Comment:
```suggestion
        controller_context.add_identity_to_log(candidate.id)
```

Reply:
Sadly `add_identity_to_log` is a [private](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/master/app/controllers/candidate_interface/candidate_interface_controller.rb#L34) method of the `CandidateInterfaceController`. What do you think about moving it above the private keyword or removing the keyword altogether, as I don't think it serves much of a purpose?

Reply:
Have moved it out of private, can revisit if opinions on this differ.

Reply:
👍